backend/src/layer/data/image_validation/image_evaluation.rs

```diff
@@ -0,0 +1,63 @@
+use crate::interface::image_validation::ImageValidationError::InvalidContent;
+use crate::interface::image_validation::Result;
+use crate::layer::data::image_validation::json_structs::SafeSearchJson;
+
+//TODO DOC
+pub struct ImageEvaluation {
+    acceptance: [u8; 5],
+}
+
+impl ImageEvaluation {
+    //TODO DOC
+    #[must_use]
+    pub const fn new(acceptance: [u8; 5]) -> Self {
+        Self { acceptance }
+    }
+
+    //TODO DOC
+    pub fn verify(&self, results: SafeSearchJson) -> Result<()> {
+        let values = result_to_arr(&results);
+        for (i, value) in values.iter().enumerate() {
+            if value > &self.acceptance[i] {
+                return Err(InvalidContent(format!(
+                    "This image contains probably {} content",
```

Comment:
message could be inside the error definition?
actual detected level my be interesting